### PR TITLE
refactor(account user agreements details): rewrite controller as class

### PR DESCRIPTION
**Pull request by antleblanc**
### Description of the Change

a0663b0

ref: #328

/cc @jleveugle @Jisay @frenautvh

### CODE REVIEW

File: client/app/account/user/agreements/details/user-agreements-details.controller.js [hunk 1]
```diff
@@ -1,54 +1,68 @@
...
-                .then((result) => {
-                    this.agreement = result[0];
-                    this.contract = result[1];
-                    this.isIndividual = result[2].legalform === "individual";
-                    this.alreadyAccepted = this.agreement.agreed === "ok";
+            return this.$q
+                .all({
+                    contractAgreement: this.fetchContractAgreement(this.$stateParams.id),
+                    contract: this.fetchContract(this.$stateParams.id),
+                    user: this.fetchUser()
```
Comment (JeremyDec) [inline client/app/account/user/agreements/details/user-agreements-details.controller.js]: For the three fetch methods, shoudn't they be declared before use ?

File: client/app/account/user/agreements/details/user-agreements-details.controller.js [hunk 2]
```diff
@@ -1,54 +1,68 @@
...
-                .catch((err) => {
-                    Alerter.error($translate.instant("user_agreement_details_error"), "agreements_details_alerter");
-                    $q.reject(err);
-                });
-        };
-
-        this.$ngInit();
-    }
-]);
+                .catch(() => this.Alerter.error(this.$translate.instant("user_agreement_details_error"), "agreements_details_alerter"));
```
Comment (JeremyDec) [inline client/app/account/user/agreements/details/user-agreements-details.controller.js]: No need to reject promise ?

File: client/app/account/user/agreements/user-agreements.controller.js [hunk 3]
```diff
@@ -1,37 +1,30 @@
...
-            Service.getList(count, offset)
+            return UserAccountAgreements
+                .getList(count, offset)
                 .then((agreements) => {
                     $scope.list = agreements;
-                }, (err) => {
-                    Alerter.error(`${$translate.instant("user_agreements_error")} ${_.get(err, "message") || err}`, "agreements_alerter");
                 })
-                .then(() => {
+                .catch((err) => Alerter.error(`${$translate.instant("user_agreements_error")} ${_.get(err, "message") || err}`, "agreements_alerter"))
```
Comment (JeremyDec) [inline client/app/account/user/agreements/user-agreements.controller.js]: No need to reject promise ?

File: client/app/account/user/agreements/details/user-agreements-details.controller.js (hunk 1 above)
Comment (antleblanc) [reply client/app/account/user/agreements/details/user-agreements-details.controller.js]: Not needed. We are in a ES6 `class`.

File: client/app/account/user/agreements/user-agreements.controller.js (hunk 3 above)
Comment (antleblanc) [reply client/app/account/user/agreements/user-agreements.controller.js]: The rejection is done on the `UserAccountAgreements` service. From here, I just want to display relevant message to the user.

**Pull request merged**